urllib3/connectionpool.py

```diff
@@ -829,6 +835,7 @@ def connection_from_url(url, **kw):
         >>> r = conn.request('GET', '/')
     """
     scheme, host, port = get_host(url)
+    port = port or port_by_scheme.get(scheme, 80)
```

Comment:
Yes.

If you pass an IPv6 literal without square brackets to httplib while passing `port=None` then it will attempt to parse a port out of the IP address, which will cause it to throw an exception. We *need* to pass a non-`None` port.